code/datums/status_effects/debuffs.dm

```diff
@@ -580,6 +567,20 @@
 	desc = "You've been zapped with something and your hands can't stop shaking! You can't seem to hold on to anything."
 	icon_state = "convulsing"
 
+//Stamina drain associated with stun batons and flashes when used on carbons
+/datum/status_effect/stamdrain
+	id = "stamdrain"
+	duration = 50
+	alert_type = null
+	status_type = STATUS_EFFECT_REFRESH
+
+/datum/status_effect/stamdrain/tick()
+	if(iscarbon(owner) && !HAS_TRAIT(owner, TRAIT_STUNRESISTANCE))
+		var/mob/living/carbon/H = owner
+		H.adjustStaminaLoss(3)
+	else
+		qdel(src)
```

Comment:
Put this in its init instead of checking it every tick.

Reply:
Ended up changing the role of stun resistance a little so this might be unnecessary.